### PR TITLE
docs: improve linking between doc sections

### LINKED ISSUE #1415: Improve reachability of the roadmap
### Clear and concise description of the problem

I knew the roadmap was there, but it took me one minute to find it in the docs.

### Suggested solution

Add an `About` link to the Guide page (and vice versa).
Add a roadmap to the `About` navbar entry.

Somewhat similar to this page:

grafik

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #1415

- Extend the `SideNav` with links to the other sections
- Retain the highlight in the nav bar, when navigating to nested pages
- Add a link to the `Roadmap` to the nav bar
- Add an `Overview` link to the API section to get back to the search page

<details>
<summary>Guide-Section</summary>

grafik

</details>

<details>
<summary>API-Section</summary>

grafik

</details>

<details>
<summary>About section</summary>

grafik

</details>

### CODE REVIEW

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -144,12 +186,14 @@ export default defineConfig({
             },
           ],
         },
+        ...otherLinks('Guide'),
```
Comment (ST-DDT) [inline docs/.vitepress/config.ts:189]: Should we append the links to the other sections or should we retain the same order on all pages?

File: docs/.vitepress/config.ts [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
+      items: [
+        {
+          text: 'Announcements',
+          link: '/about/announcements',
+        },
+        {
+          text: 'Roadmap',
+          link: '/about/roadmap/',
+        },
+      ],
```
Comment (ST-DDT) [inline docs/.vitepress/config.ts:41]: Should we also include the `Team` page here?

Comment (Shinigami92) [reply docs/.vitepress/config.ts:41]: yes

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (Shinigami92) [reply docs/.vitepress/config.ts:189]: how do other pages/docs handle this?

Comment (ST-DDT) [reply docs/.vitepress/config.ts:189]: Vitest retains the order, but they don't use sections as much as we do.
I don't know any other similar docs pages.

Comment (ST-DDT) [reply docs/.vitepress/config.ts]: Done

File: docs/.vitepress/config.ts (hunk 2 above)
Comment (ST-DDT) [reply docs/.vitepress/config.ts:47]: Done

File: docs/.vitepress/config.ts [hunk 3]
```diff
@@ -1,11 +1,56 @@
...
 import { apiPages } from './api-pages';
 import { currentVersion, oldVersions } from './versions';
 
+type SidebarGroup = DefaultTheme.SidebarGroup;
+
 const description =
   'Generate massive amounts of fake (but reasonable) data for testing and development.';
 const image = 'https://fakerjs.dev/social-image.png';
 
+function extendSideNav(current: SidebarGroup): SidebarGroup[] {
```
Comment (ST-DDT) [inline docs/.vitepress/config.ts:12]: As an alternative, we could use the full side nav everywhere, but collapse all sections not belonging to the current one.

**Review by Shinigami92 — APPROVED** [review]
I havn't checked it and turst your screenshots
But what I can see on the screenshots is very nice :ok_hand:

**Review by ejcheng — APPROVED** [review]

**Comment by Shinigami92** [comment]
Post-Tested with real docs, nice work :raised_hands:

**Pull request merged**